### ISSUE TITLE
Outdated deps: Incorrect analysis

### ISSUE THREAD

**Issue by rrrene** [hexfaktor/hex_faktor_web#2]
http://localhost:4000/github/inch-ci/Hello-World-Elixir
![image](https://cloud.githubusercontent.com/assets/311914/13036912/21171a38-d374-11e5-8844-3a91f6cfb52c.png)

The requirement is exact: version `0.4.0-dev` should be loaded. The locked version matches that, but the severity is `update` nonetheless.

**Comment by danielberkompas** [comment]
Another case: https://beta.hexfaktor.org/github/infinitered/phoenix_base

![hexfaktor](https://cloud.githubusercontent.com/assets/1186992/14605896/963d2d06-052f-11e6-8a0a-e07b30c468d2.png)

I'm pretty sure the `~2.0` version requirement allows for the latest version of `2.0.1`.

**Comment by danielberkompas** [comment]
Also here: https://beta.hexfaktor.org/github/danielberkompas/cloak

![hexfaktor](https://cloud.githubusercontent.com/assets/1186992/14605923/bfa3c9ac-052f-11e6-89ad-3cf614d48ceb.png)

The `>= 1.0.0` should match the latest version, `1.1.5`.

**Comment by rrrene** [comment]
:+1: will fix!

**Comment by rrrene** [comment]
Hi Daniel,

this issue should now be resolved. I fixed the bug and ran all projects against the new code.

**Comment by marcoherbst** [comment]
Loving it, but I would like our developers to get something more sympathetic than an orange light for using future versions.... how about purple ?? :) ....

![hex](https://cloud.githubusercontent.com/assets/5648894/15558174/02be7e2c-22d0-11e6-8ae8-19b2d8cf71f6.png)

On Gemnasium, they seem to use green.

**Comment by rrrene** [comment]
**TL;DR** Are you unhappy with the methodology of separating prod/other dependencies, or just the color? Shouldn't we have a "warning" color for outdated prod dependencies?

Haha, that's one big screenshot :D Okay, let's discuss!

Actually we are using green for new versions still fulfilling the requirement (just like Gemnasium). 

In your example `httpoison` is required to be `~> 0.8.2`. In your `mix.lock` it is locked to `0.8.2` and there ist a newer version `0.8.3`. **The dep is green** nonetheless, since the newest version still satisfies your requirement. With `calendar`, the newest version (`0.14.1`) does no longer fullfil the requirement (`~> 0.12.4`), therefore you get a "warning".

Now about the colors: Gemnasium gives you a yellow indicator when new deps are released and has different lists for runtime and development environments. With HexFaktor I did not want to recreate that separation, but at the same time I feel that production dependencies are far more important to keep up-to-date than development-related.

If you don't update Credo for six months you're probably missing out on bug fixes and new checks, but who cares? If you don't update a production dep (like say Timex or HtmlSanitizeEx) for six months, there will probably have been two minor version bumps and seamingly half the public API has changed.

Therefore I wanted to use two distinct colors for prod and other deps (orange and yellow; we will also have red in the future, for security issues). 

Thx for caring about this! :+1: So, to start a discussion, under which circumstances would you want to see a less aggressive color for outdated production dependencies no longer satisfying their stated requirement?

**Comment by marcoherbst** [comment]
Sorry, I wrote that message poorly, the one I was talking about was phoenix-ecto ... where we require 3.0.0-beta and the current is 2.0.2 , I should have highlighted that.

**Comment by rrrene** [comment]
Ah, okay! I saw that too and created an issue this morning: #15 It's simply a bug! :grinning:

**Comment by rrrene** [comment]
@marcoherbst This should now be fixed for new builds. I re-ran the analysis on https://beta.hexfaktor.org/github/evercam/evercam-media manually. Seems good to me :+1:

**Comment by marcoherbst** [comment]
Beautiful Rene - nice one :)
